### ISSUE TITLE
Incomplete tests: Cert rotation/issuance failure tolerance

### ISSUE THREAD

**Issue by hklai** [istio/old_auth_repo#288]
Context: [production readiness proposal and plan](https://goo.gl/pmjuYg)

The feature “_Cert rotation/issuance failure tolerance_” is identified as incomplete test coverage. As a measure to reach Istio production quality in 0.3, this issue is opened to track the progress of providing sufficient test coverage for the feature in question.

This issue is initially assigned to the working group leads.

**Recommendations:**

- Come up with a test plan for how the feature should be tested. Identify and document (in this issue, or a separate document this issue will link to) test cases and whether each will be covered by an integration test (with some level of mocking and/or starting up selected components in test environment), or end to end test (aka system test, which brings up the whole Istio to exercise the feature; currently live in the istio/istio repo).
- If you expect the test plan is complex and controversial, consider having it formally [reviewed](http://goo.gl/pRuXqe).
- State an ETA in the issue, and assign this to a developer who will execute the test plan.
- Execute the test plan as the highest priority for 0.3.
- For any bug or defect identified by the new test cases, file a new issue and link it from this one.
Close this issue only when the test plan is completely executed, and all additional issues identified in step 3 are fixed.

If you see a risk of not fixing this issue by end of 0.3, please contact istio-production-quality@googlegroups.com as soon as possible. We will also periodically ping the issue for update.

**Comment by myidpt** [comment]
This will be done through:
1. Write tests to cover cert rotation.
2. Improve logging of Istio CA to help track the cert issuance failures and CA status.
3. For HA, have doc to instruct deploying multiple CAs. Ensure that multiple Istio CAs can coexist in a cluster without racing.

**Comment by myidpt** [comment]
Update: I have manually tested that 3) works. The istio.io doc will have instructions about bringing up multiple CAs to enhance HA.
(1) will be done by extending the current Istio security integration test. This was blocked by the repo migration and the PR will be sent out within a week.
(2) will be done within 1-2 weeks.

**Comment by wattli** [comment]
sg, thanks Oliver!

**Comment by myidpt** [comment]
@mangchiandjjoe Jae is helping with (1) the security integration test now. Thanks, Jae!

**Comment by mangchiandjjoe** [comment]
Create a PR for certificates rotation https://github.com/istio/istio/pull/1739
